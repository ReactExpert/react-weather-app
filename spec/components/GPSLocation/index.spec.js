import React from 'react';
import GPSLocation from '../../../src/components/GPSLocation';
import { enzymeConfig, mount } from '../../enzymeConfig';

enzymeConfig();

describe('GPSLocation', () => {
  it('matches snapshot', () => {
    const component = mount(<GPSLocation onGPSLocationClick={() => { }} />);

    expect(component).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls animate, getClass', (done) => {
      const component = mount(<GPSLocation onGPSLocationClick={() => { }} />);
      const instance = component.instance();

      instance.animate = jest.fn();

      setTimeout(() => {
        expect(instance.animate).toBeCalled();
        done();
      }, 500);
    });
  });

  describe('onMouseDown', () => {
    it('triggers mousedown', () => {
      spyOn(GPSLocation.prototype, 'onMouseDown');

      const component = mount(<GPSLocation updating={false} onGPSLocationClick={() => { }} />);

      component.find('.gps-location').simulate('mousedown');

      expect(GPSLocation.prototype.onMouseDown).toBeCalled();
    });

    it('adds class press to button', (done) => {
      const component = mount(<GPSLocation updating={false} onGPSLocationClick={() => { }} />);
      const instance = component.instance();

      component.find('.gps-location').simulate('mousedown');

      setTimeout(() => {
        expect(instance.button.current.className).toContain('press');
        done();
      }, 100);
    });
  });

  describe('onMouseUp', () => {
    it('triggers onMouseUp', () => {
      spyOn(GPSLocation.prototype, 'onMouseUp');

      const component = mount(<GPSLocation updating={false} onGPSLocationClick={() => { }} />);

      component.find('.gps-location').simulate('mouseup');

      expect(GPSLocation.prototype.onMouseUp).toBeCalled();
    });

    it('removes class press from button', (done) => {
      const component = mount(<GPSLocation updating={false} onGPSLocationClick={() => { }} />);
      const instance = component.instance();

      component.find('.gps-location').simulate('mousedown');
      component.find('.gps-location').simulate('mouseup');

      setTimeout(() => {
        expect(instance.button.current.className).not.toContain('press');
        done();
      }, 100);
    });
  });

  describe('onTouchStart', () => {
    it('triggers onMouseDown', () => {
      spyOn(GPSLocation.prototype, 'onMouseDown');

      const component = mount(<GPSLocation updating={false} onGPSLocationClick={() => { }} />);

      component.find('.gps-location').simulate('touchstart');

      expect(GPSLocation.prototype.onMouseDown).toBeCalled();
    });
  });

  describe('onTouchEnd', () => {
    it('triggers onMouseUp', () => {
      spyOn(GPSLocation.prototype, 'onMouseUp');

      const component = mount(<GPSLocation updating={false} onGPSLocationClick={() => { }} />);

      component.find('.gps-location').simulate('touchend');

      expect(GPSLocation.prototype.onMouseUp).toBeCalled();
    });
  });
});
