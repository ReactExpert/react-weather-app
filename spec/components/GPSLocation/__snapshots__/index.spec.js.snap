// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GPSLocation matches snapshot 1`] = `
<GPSLocation
  onGPSLocationClick={[Function]}
>
  <button
    aria-label="Get current location"
    className="gps-location"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <svg
      id="gpslocation"
      viewBox="0 0 96 96"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        id="XMLID_1_"
      >
        <circle
          cx="48"
          cy="48"
          id="XMLID_3_"
          r="9.8"
        />
        <path
          d="M80.5 44.8h-6.7C72.3 33 63 23.7 51.3 22.2v-6.7h-6.5v6.7C33 23.7 23.7 33 22.2 44.8h-6.7v6.5h6.7C23.7 63 33 72.3 44.8 73.8v6.7h6.5v-6.7C63 72.3 72.3 63 73.8 51.3h6.7v-6.5zM48 67.5c-10.8 0-19.5-8.7-19.5-19.5S37.2 28.5 48 28.5 67.5 37.2 67.5 48 58.8 67.5 48 67.5z"
          id="XMLID_6_"
        />
      </g>
    </svg>
  </button>
</GPSLocation>
`;
